### ISSUE TITLE
Post Date block doesn't show the modified date for newly published posts

### ISSUE THREAD

**Issue by bugnumber9** [WordPress/gutenberg#57915]
### Description

When a new post is published and hasn't been modified after publishing, the Post Date block doesn't show the modified date.

I confirmed that `get_the_modified_date` function works as expected in a non-FSE theme (Hello theme) by altering the code in single.php as shown here:
![hello](https://github.com/WordPress/gutenberg/assets/1769548/3aa7b431-5f6b-4bde-8327-25137de27bf6)
The output is expected:
![frontend2](https://github.com/WordPress/gutenberg/assets/1769548/0bfb17d8-4b84-41dc-9893-d1f9b709b215)

However, when I change a single post template in site editor (standard 2024 theme) as shown here:
![site-editor](https://github.com/WordPress/gutenberg/assets/1769548/a237969c-513a-4efb-a14a-98485e9c1dfa)
The modified date on the frontend is empty:
![frontend](https://github.com/WordPress/gutenberg/assets/1769548/1bf9e89f-1576-4435-a9d9-444bdb1bf25a)

### Step-by-step reproduction instructions

Shown above.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by benniledl** [comment]
I see, that this was the most recent change in the post date block and was intentional: #46839

**Comment by benniledl** [comment]
@bugnumber9 I recommend using 
`get_the_modified_date() ? get_the_modified_date() : get_the_date();`
This returns the modified date and if that is empty it returns the normal date

**Comment by bugnumber9** [comment]
Well, this is a bit weird.
Let's say, I want to show the "last updated" not "created" date in the postmeta section (using FSE) - it won't show for posts that weren't modified after publishing.
I was able to create a workaround using Block Logic and a custom PHP function, but it's surely not ideal.

**Comment by benniledl** [comment]
@bugnumber9 you don't need an extra function you can just do 
`echo 'Modified: ' .  get_the_modified_date() ? get_the_modified_date() : get_the_date();`

**Comment by bugnumber9** [comment]
> @bugnumber9 I recommend using `get_the_modified_date() ? get_the_modified_date() : get_the_date();` This returns the modified date and if that is empty it returns the normal date

I used Block Logic plugin and a custom PHP function to "fix" this. But I'm using a FSE theme and building postmeta section in site editor, so not being able to use Post Date block with "show modified date" option because of this behavior is kinda weird.

**Comment by bugnumber9** [comment]
> @bugnumber9 you don't need an extra function you can just do `echo 'Modified: ' . get_the_modified_date() ? get_the_modified_date() : get_the_date();`

I'm using FSE theme and I use Post Date block in postmeta template.

Besides that, both `get_the_date` and  `get_the_modified_date` functions always return dates, so in PHP I can just do `echo 'Modified: ' . get_the_modified_date();` and it'll **always** work.

It's the Post Date block behavior that I find weird in this case because I can't reliably use `<!-- wp:post-date {"displayType":"modified"} /-->` where I need to.

**Comment by carolinan** [comment]
We need to think about how can we satisfy both these requests without making the block options too complex.

**Comment by bugnumber9** [comment]
@carolinan IMO the block should behave just like `get_the_date` and `get_the_modified_date` functions.
In other words, `<!-- wp:post-date {"displayType":"modified"} /-->` should always return the date.
If I make a decision to use two date blocks to display published and modified dates - I realize they may be the same in some cases. So this now falls into the block display logic realm.

Besides that, the current behavior will still display two identical dates if the post was edited, say, several minutes after publishing which kinda defeats the purpose of https://github.com/WordPress/gutenberg/pull/46839 
While there's no native block display logic functionality in Gutenberg, 3rs party solutions exist and this is where the show/hide logic should reside.

**Comment by carolinan** [comment]
And that is your opinion, please respect that there are other users who have expressed different needs.

**Comment by bugnumber9** [comment]
> And that is your opinion, please respect that there are other users who have expressed different needs.

Yes, absolutely. It's just my opinion of course, and I was able to achieve the behavior I wanted using a 3rd party block display logic plugin to fall back to published date.
All I'm saying is that I find the current behavior somewhat weird (again, maybe it's just me) because it a) doesn't completely prevent showing 2 identical dates, and b) makes it impossible to use `<!-- wp:post-date {"displayType":"modified"} /-->` if I want to only show the modified date as it'll be empty in some cases.

That being said, the display logic of both published and modified dates is often a matter of personal preferences or project requirements.
E.g. one of the projects I've built required that modified date is only publicly displayed if the difference between published and modified dates is 4 weeks or more yet is always visible to the editorial team. This is easy to do in the non-FSE theme in several possible ways, but that would usually be either hard code or ACF-like approach.
I can't imagine a reasonably simple way to move the display logic to block controls and satisfy varying needs, thus mentioned that IMO this falls into block display logic realm rather than block behavior or controls.

Just my 2.5 cents :)

**Comment by carolinan** [comment]
I personally don't find it strange that the **modified** date literally only shows if the post has been **modified**.

**Comment by graylaurenm** [comment]
I've run into this issue as well. It's specifically a problem if I want to have a "modified" label, but then a date doesn't show up. We've partially solved this by creating a plugin that forces either the modified or published date to show: https://www.smallplugins.com/get-started/post-date-smarter-modified/. However, I think it would be really helpful to have a 'label' field, similar to the author block, so that if the modified date doesn't show, neither does the label for it.

**Comment by matthewbullweb** [comment]
I see an issue for posts updated the same day. 
 
Modified is "202501230125" get_the_modified_date( 'Ymdhi', $post_ID )
Created is "202501230125" get_the_date( 'Ymdhi', $post_ID )

Modified should read  "202501231325" to be shown. Also feel comparison should be >= above or equal too.

It is a wp core bug https://github.com/WordPress/WordPress/blob/8eb3443f3194f4d04a8dadf90a499b497dfb0ecf/wp-includes/blocks/post-date.php#L52

**Comment by gchriz** [comment]
> I see an issue for posts updated the same day.
> 
> Modified is "202501230125" get_the_modified_date( 'Ymdhi', $post_ID ) Created is "202501230125" get_the_date( 'Ymdhi', $post_ID )
> 
> Modified should read "202501231325" to be shown. Also feel comparison should be >= above or equal too.

I stumbled on this yesterday. My bug report with fix: See https://core.trac.wordpress.org/ticket/63834
